eclipse-collections/src/main/java/org/eclipse/collections/impl/map/sorted/mutable/MutableSortedMapFactoryImpl.java

```diff
@@ -27,6 +27,12 @@ public <K, V> MutableSortedMap<K, V> empty()
         return TreeSortedMap.newMap();
     }
 
+    @Override
+    public <K, V> MutableSortedMap<K, V> empty(Comparator<? super K> comparator)
+    {
```

Comment:
@Desislav-Petrov Please move the code from `with(Comparator)` to here with the null check for comparator and change with(Comparator) to call empty().